tests/forms_tests/field_tests/test_urlfield.py

```diff
@@ -7,150 +7,117 @@
-        self.assertEqual('http://valid-----hyphens.com', f.clean('http://valid-----hyphens.com'))
-        self.assertEqual(
-            'http://some.idn.xyz\xe4\xf6\xfc\xdfabc.domain.com:123/blah',
-            f.clean('http://some.idn.xyzäöüßabc.domain.com:123/blah')
-        )
-        self.assertEqual(
-            'http://www.example.com/s/http://code.djangoproject.com/ticket/13804',
-            f.clean('www.example.com/s/http://code.djangoproject.com/ticket/13804')
-        )
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('[a')
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://[a')
-
-    def test_url_regex_ticket11198(self):
-        f = URLField()
-        # hangs "forever" if catastrophic backtracking in ticket:#11198 not fixed
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://%s' % ("X" * 200,))
 
-        # a second test, to make sure the problem is really addressed, even on
-        # domains that don't fail the domain label length check in the regex
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://%s' % ("X" * 60,))
-
-    def test_urlfield_2(self):
-        f = URLField(required=False)
-        self.assertEqual('', f.clean(''))
-        self.assertEqual('', f.clean(None))
-        self.assertEqual('http://example.com', f.clean('http://example.com'))
-        self.assertEqual('http://www.example.com', f.clean('http://www.example.com'))
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('foo')
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://')
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://example')
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://example.')
-        with self.assertRaisesMessage(ValidationError, "'Enter a valid URL.'"):
-            f.clean('http://.com')
-
-    def test_urlfield_5(self):
+    def test_urlfield_widget_max_min_length(self):
         f = URLField(min_length=15, max_length=20)
-        self.assertWidgetRendersTo(f, '<input id="id_f" type="url" name="f" maxlength="20" minlength="15" required>')
-        with self.assertRaisesMessage(ValidationError, "'Ensure this value has at least 15 characters (it has 12).'"):
+        self.assertWidgetRendersTo(
+            f,
+            '<input id="id_f" type="url" name="f" maxlength="20" minlength="15" required>',
+        )
+        msg = "'Ensure this value has at least 15 characters (it has 12).'"
+        with self.assertRaisesMessage(ValidationError, msg):
             f.clean('http://f.com')
+        msg = "'Ensure this value has at most 20 characters (it has 37).'"
```

Comment:
I would move it right before the `with self.assertRaisesMessage(ValidationError, msg):`

Reply:
Changed :+1: